File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        vowels="aeiou"
        count,maxcount=0,0
        for i in range(len(s)):
            if s[i] in vowels:
                count+=1
            if i>=k:
                if s[i-k] in vowels:
                    count-=1
            maxcount=max(count,maxcount)
        return maxcount

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int count=0;
        int maxcount=0;
        String vowels = "aeiou";

        for(int i=0;i<s.length();i++){
            //add new char
            if(vowels.indexOf(s.charAt(i))!=-1){
                count++;
            }
            //remove char leaving window
            if(i>=k){
                if(vowels.indexOf(s.charAt(i-k))!=-1){
                    count--;
                }
            }
            maxcount=Math.max(maxcount,count);
        }
        return maxcount;
    }
}